stl-decomp-4j/src/main/java/com/github/servicenow/ds/stats/stl/SeasonalTrendLoess.java

```diff
@@ -446,7 +461,7 @@ private void sanityCheck(double[] data) {
 		fRobustIterations = no;
 
 		fLoessSmootherFactory = new LoessSmoother.Builder() //
-				.setWidth(fTrendSettings.getWidth()) //
+				.setWidth(fTrendSettings.getWidth() + (fExogenousData == null ? 0 : fExogenousData.length)) //
```

Comment:
This length determines the data points used in the local least-squares fit, that is the no. of rows in the regressor matrix. Since exog.inputs increase the no. of columns of this regressor, the problem may get underdetermined (or become less-overdetermined), so also increasing the data points proportionally keeps the difference in no. of rows and columns the same.